deepdrivemd/sim/openmm_stream/run_openmm.py

```diff
@@ -0,0 +1,198 @@
+            sim.context.setVelocities(velocities)
+        else:
+            print("Generating velocities randomly")
+            sim.context.setVelocitiesToTemperature(
+                cfg.temperature_kelvin * u.kelvin, random.randint(1, 10000)
+            )
+        return True
+    else:
+        print("There are no outliers")
+        return False
+
+
+def init_input(cfg):
+    """
+    The first iteration of the simulation is initialized from pdb
+    files in cfg.initial_pdb_dir. For the given simulation the pdb file is
+    selected using simulation task_id in a round robin fashion.
+    """
+    pdb_files = list(cfg.initial_pdb_dir.glob("*.pdb"))
+    pdb_files.sort()
+    n = len(pdb_files)
+    i = int(cfg.task_idx) % n
+    cfg.pdb_file = pdb_files[i]
+    print(f"init_input: n = {n}, i = {i}, pdb_file = {cfg.pdb_file}")
+
+
+def run_simulation(cfg: OpenMMConfig):
+    init_input(cfg)
+
+    # openmm typed variables
+    dt_ps = cfg.dt_ps * u.picoseconds
+    report_interval_ps = cfg.report_interval_ps * u.picoseconds
+    simulation_length_ns = cfg.simulation_length_ns * u.nanoseconds
+    temperature_kelvin = cfg.temperature_kelvin * u.kelvin
+
+    # Handle files
+    with Timer("molecular_dynamics_SimulationContext"):
+        ctx = SimulationContext(cfg)
+
+    # Create openmm simulation object
+    with Timer("molecular_dynamics_configure_simulation"):
+        sim = configure_simulation(
+            pdb_file=ctx.pdb_file,
+            top_file=ctx.top_file,
+            solvent_type=cfg.solvent_type,
+            gpu_index=0,
+            dt_ps=dt_ps,
+            temperature_kelvin=temperature_kelvin,
+            heat_bath_friction_coef=cfg.heat_bath_friction_coef,
+        )
+
+    # Number of steps to run each simulation
+    nsteps = int(simulation_length_ns / dt_ps)
+
+    report_steps = int(report_interval_ps / dt_ps)
+    print("report_steps = ", report_steps)
+
+    # Configure reporters to write output files
+    with Timer("molecular_dynamics_configure_reporters"):
+        configure_reporters(sim, ctx, cfg, report_steps)
+
+    # Infinite simulation loop
+    for iteration in itertools.count(0):
+        # Run simulation for nsteps
+        print(f"Simulation iteration {iteration}")
+        sys.stdout.flush()
+        with Timer("molecular_dynamics_step"):
+            sim.step(nsteps)
+        prepare_simulation(cfg, iteration, sim)
+
+
+def adios_configuration(cfg: OpenMMConfig):
+    """
+    Read a template adios.xml file, replace "SimulationOutput"
+    stream name with the simulation taskid and write the resulting
+    configuration file into simulation directory.
+    """
+    adios_cfg = cfg.output_path / "adios.xml"
+    shutil.copy(cfg.adios_xml_sim, adios_cfg)
+    cfg.adios_cfg = adios_cfg
+    taskdir = os.path.basename(cfg.output_path)
+    with open(cfg.adios_cfg, "r") as f:
+        textxml = f.read()
+    textxml = textxml.replace("SimulationOutput", taskdir)
+    f = open(cfg.adios_cfg, "w")
```

Comment:
Sorry, accidentally closed this one. It should also use a context manager `with open() as f: ...`

Reply:
Done